system.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package systems
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/juju/errors"
+
+	"github.com/juju/systems/channel"
+)
+
+// System represents an OS/Channel or Resource.
+// Systems can also be converted to and from a series string.
+type System struct {
+	OS       string          `json:"os,omitempty"`
+	Channel  channel.Channel `json:"channel,omitempty"`
+	Resource string          `json:"resource,omitempty"`
+}
+
+// Validate returns with no error when the System is valid.
+func (s System) Validate() error {
+	if s.OS != "" {
+		if s.Resource != "" {
+			return errors.NotValidf("resource cannot be specified with os")
+		}
+		if !validOS.Contains(s.OS) {
+			return errors.NotValidf("os %q", s.OS)
+		}
+		if s.Channel == channel.Empty {
+			return errors.NotValidf("missing channel")
+		}
+	} else if s.Resource != "" {
+		if s.Channel != channel.Empty {
+			return errors.NotValidf("channel cannot be specified with resource")
+		}
+	} else {
+		return errors.NotValidf("one of os or resource must be specified")
+	}
```

Comment:
Maybe do this check first,
```
if s.OS == "" && s.Resource == "" {
    return errors.NotValidf("one of os or resource must be specified")	
}
```

then the rest....

Maybe makes it easier to read?